static/js/server_events_dispatch.js

```diff
@@ -187,18 +187,15 @@ exports.dispatch_normal_event = function dispatch_normal_event(event) {
     case 'realm_bot':
         if (event.op === 'add') {
             bot_data.add(event.bot);
-            settings_users.update_user_data(event.bot.user_id, event.bot);
         } else if (event.op === 'remove') {
             bot_data.deactivate(event.bot.user_id);
             event.bot.is_active = false;
-            settings_users.update_user_data(event.bot.user_id, event.bot);
         } else if (event.op === 'delete') {
```

Comment:
I don't think your changes make anything worse here, but the 'delete' situation is tricky, since I don't think know if we have a way to delete the row easily?  and do we want to just update it somehow?  and if we update it, what happens when `get_item` fails here?

Reply:
Resolved in chat, we explicitly ignore this now.